lib/install.js

```diff
@@ -235,7 +260,12 @@ function install(gyp, argv, callback) {
                     if (err) {
                         after_place(err);
                     } else {
-                        place_binary(from,to,opts,after_place);
+                        var fileName = from.startsWith('file://') && from.replace(/^file:\/\//, '');
+                        if (fileName) {
+                            extract_from_local(fileName, to, callback);
```

Comment:
You're passing `callback` here, but I think you meant `after_place`. Right now, if the `extract_from_local` function fails, (and it can, with file not found for instance), it'll not potentially fallback to building from source, which is the role of `after_place`.